### PR TITLE
Georgia adjusted gross income

### LINKED ISSUE #2882: Georgia AGI
ga_addtions
ga_subtractions
ga_agi

### PR DESCRIPTION
**Pull request by Eleanor-Zhu**
### Summary
📄🧮🇺🇸

This pull request adds new files and variables to implement the Georgia state income tax rules for calculating AGI and exempting military retirement income. It reuses an existing file for the OASDI subtraction for Georgia and adds test cases for the military retirement income tax exemption. It modifies the `policyengine_us` parameters, variables, and tests directories.

> _We're the masters of the tax code, we know the rules of AGI_
> _We add and subtract the income, we calculate the exemptions_
> _We test the logic and the numbers, we write the files in YAML and Python_
> _We're the policy engine of Georgia, we're the `policyengine_us`_

### Walkthrough
*  Create new variable `ga_agi` that calculates Georgia AGI based on federal AGI and state-specific additions and subtractions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-2a01e938f81b57852fbef5fc4ebf1f1f1f686d6dd1427ebffee6e5deb5eceda3L1-R15))
*  Add new file `policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml` that defines a list of Georgia AGI subtractions from federal AGI ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-78b003d14e0343c80f348f0e7c9179f484b6d9cde077d55b233847400913fa5eL1-R42))
*  Add new variable `ga_agi_subtractions` that aggregates the state-specific subtractions from federal AGI ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-26fa4ea46a239024fdbb753b9525d97ad128aa58d5e64773f8f1763188440062L1-R15))
*  Add new variable `ga_agi_additions` that aggregates the state-specific additions to federal AGI ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-bc47cea7e63e16705cb4ddc84c44090fc4e7e4e5fcf17498e0603507bd7f5adfL1-R13))
*  Create new variable `ga_military_retirement_income_tax_exemption` that calculates the exemption amount for military retirement income based on age ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-b93276b7803903845b5d481d1c591d455f97d02d0811fb8d568a357daadbef56R1-R14))
*  Add new file `policyengine_us/parameters/gov/states/ga/tax/income/exemptions/military_retirement/age_threshold.yaml` that defines the age range and amount for the military retirement income tax exemption ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-58c52db6d61e88940d70d8902cdc97ff6cda9a08118bddd84562507b7ed8947eR1-R22))
*  Add test cases for the `ga_military_retirement_income_tax_exemption` variable in `policyengine_us/tests/policy/baseline/gov/states/ga/tax/income/exemptions/ga_military_retirement_income_tax_exemption.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-c76a8f6f8fae0603c67058f0c421682fc59405469c2079d2b3bb95b107f15d05L1-R38))
*  Modify existing file `policyengine_us/parameters/gov/states/ks/tax/income/agi/subtractions/oasdi/agi_limit.yaml` to change the state code, label, and reference from KS to GA and reuse the file for the taxable OASDI subtraction for Georgia ([link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-6641faa914b569aad2cfc0ea9d586e867d1b91cf2fc2180543b7e11321e08524L1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-6641faa914b569aad2cfc0ea9d586e867d1b91cf2fc2180543b7e11321e08524L8-R8), [link](https://github.com/PolicyEngine/policyengine-us/pull/2962/files?diff=unified&w=0#diff-6641faa914b569aad2cfc0ea9d586e867d1b91cf2fc2180543b7e11321e08524R16))

fixes #2882

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:1]: need changelog_entry

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+description: List of California AGI subtractions.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:1]: change description

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # disabled first responder
+    # firefighter, payments received
+    # For a firefighter, an amount equal to 100% of any premium paid
+    # Income received by a surviving family member
+    # certain disaster relief payments
+    # For owners of a pass-through entity or entities   
+
+metadata:
+  reference:
+    - title: 2022 Form 500 Instructions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:38]: add legal code reference

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14
+    - title: 2021 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2022/01/2021_IT-511_Individual_Income_Tax_Booklet-12.29.21-Web-Version.pdf#page=14
+  period: YEAR
+  label: list of Georgia AGI subtractions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:43]:
```suggestion
  label: Georgia adjusted gross income subtractions
```

File: policyengine_us/parameters/gov/states/ga/tax/income/exemptions/military_retirement/age_threshold.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+description: Age range defining someone eligible for Georgia military retirement income tax exemption
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/exemptions/military_retirement/age_threshold.yaml:1]: change to format of: Georgia does xyz

File: policyengine_us/parameters/gov/states/ks/tax/income/agi/subtractions/oasdi/agi_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/agi/subtractions/oasdi/agi_limit.yaml:1]: change back

File: policyengine_us/variables/gov/states/ga/ga_agi.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class ga_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia AGI"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ga/ga_agi.py:7]:
```suggestion
    label = "Georgia adjusted gross income"
```

File: policyengine_us/parameters/gov/states/ga/tax/income/exemptions/military_retirement/age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/exemptions/military_retirement/age_threshold.yaml:1]: rename to amount.yaml

File: policyengine_us/variables/gov/states/ga/tax/income/exemptions/military_retirement_income_tax_exemption.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ga/tax/income/exemptions/military_retirement_income_tax_exemption.py:1]: delete file

File: policyengine_us/tests/policy/baseline/gov/states/ga/tax/income/exemptions/ga_military_retirement_income_tax_exemption.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ga/tax/income/exemptions/ga_military_retirement_income_tax_exemption.yaml:1]: delete file

File: policyengine_us/variables/gov/states/ga/ga_agi_additions.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from policyengine_us.model_api import *
+
+
+class ga_agi_additions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia AGI additions to federal AGI"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ga/ga_agi_additions.py:7]:
```suggestion
    label = "Georgia additions to federal adjusted gross income"
```

File: policyengine_us/variables/gov/states/ga/ga_agi_subtractions.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class ga_agi_subtractions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia AGI subtractions from federal AGI"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ga/ga_agi_subtractions.py:7]:
```suggestion
    label = "Georgia subtractions from federal adjusted gross income"
```

File: changelog_entry.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: minor
```

File: changelog_entry.yaml [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+  changes:
+    added:
+      - Legal code reference for GA Tax Credits
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
      - Georgia adjusted gross income structure.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+metadata:
+  reference:
+    - title: 2022 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14
+    - title: 2021 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2022/01/2021_IT-511_Individual_Income_Tax_Booklet-12.29.21-Web-Version.pdf#page=14
+    - title: Official Code of Georgia Annotated | CH7 Income Taxes
+      href: https://advance.lexis.com/container/?pdmfid=1000516&crid=1912cdf2-1200-4d14-97ab-12dbba46d7fc&pdtocsearchterm=Subtractions&pdtocsearchoption=docsonly&pdsearchterms=&pdtypeofsearch=TOCSearchDoc&pdfilterstring=MTA5MTIwMw&pdsearchdisplaytext=Official+Code+of+Georgia+Annotated&pdcontextvalue=statutes-legislation&pdtocfullpath=%2Fshared%2Ftableofcontents%2Furn%3AcontentItem%3A63RH-PW33-CH1B-T4TR-00008-00&pdbcts=1694467123349&config=00JABlN2Q2OTIwYi1kMjQwLTQxMWEtOWM1YS00MzUwY2MzYjQ5ZTAKAFBvZENhdGFsb2eiEwC2ZWq2J6k0Uwbdk8jZ&ecomp=7gf5kkk&prid=cfbd617a-9be3-4e6c-8a31-29248e6c555b
+  period: YEAR
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:44]:
```suggestion
  period: year
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    # firefighter, payments received
+    # For a firefighter, an amount equal to 100% of any premium paid
+    # Income received by a surviving family member
+    # certain disaster relief payments
+    # For owners of a pass-through entity or entities   
+
+metadata:
+  reference:
+    - title: 2022 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14
```
Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:39]: the link seems invalid

Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:39]: for the same link in all files

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    - title: 2021 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2022/01/2021_IT-511_Individual_Income_Tax_Booklet-12.29.21-Web-Version.pdf#page=14
```
Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:41]: the link seems invalid (couldn't go through)

Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:41]: for the same link in all files

File: changelog_entry.yaml (hunk 9 above)
Comment (Kevintowne) [inline changelog_entry.yaml]:
```suggestion
      - Legal code reference for Georgia Tax Credits
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 13]
```diff
@@ -0,0 +1,46 @@
+description: List of Georgia AGI subtractions.
```
Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:1]:
```suggestion
description: List of Georgia adjusted gross income subtractions.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    # certain disaster relief payments
+    # For owners of a pass-through entity or entities   
+
+metadata:
+  reference:
+    - title: 2022 Form 500 Instructions
+      href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14
+    - title: 2021 Form 500 Instructions
+      href: https://www.zillionforms.com/2021/I2122607361.PDF
+    - title: Official Code of Georgia Annotated | CH7 Income Taxes
```
Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:42]: could add the correponding subchapter number after 'CH7 Income Taxes' which the link direct to

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml [hunk 15]
```diff
@@ -8,7 +8,7 @@ metadata:
     - title: 2022 Form 500 Instructions
       href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=12
     - title: 2021 Form 500 Instructions
-      href: https://houpl.org/wp-content/uploads/2022/01/2021_IT-511_Individual_Income_Tax_Booklet-12.29.21-Web-Version.pdf#page=13
+      href: https://www.zillionforms.com/2021/I2122607361.PDF
```
Comment (Kevintowne) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml:11]: add the page number

File: policyengine_us/variables/gov/states/ga/ga_agi.py [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+
+class ga_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia adjusted gross income"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14"
+        "https://www.zillionforms.com/2021/I2122607361.PDF"
```
Comment (Kevintowne) [inline policyengine_us/variables/gov/states/ga/ga_agi.py:12]: add the page number

File: policyengine_us/variables/gov/states/ga/ga_agi_additions.py [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+
+class ga_agi_additions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia additions to federal adjusted gross income"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14"
+        "https://www.zillionforms.com/2021/I2122607361.PDF"
```
Comment (Kevintowne) [inline policyengine_us/variables/gov/states/ga/ga_agi_additions.py:12]: add the page number

File: policyengine_us/variables/gov/states/ga/ga_agi_subtractions.py [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+class ga_agi_subtractions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia subtractions from federal adjusted gross income"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14"
+        "https://www.zillionforms.com/2021/I2122607361.PDF"
```
Comment (Kevintowne) [inline policyengine_us/variables/gov/states/ga/ga_agi_subtractions.py:12]: add the page number

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+      href: https://advance.lexis.com/container/?pdmfid=1000516&crid=1912cdf2-1200-4d14-97ab-12dbba46d7fc&pdtocsearchterm=Subtractions&pdtocsearchoption=docsonly&pdsearchterms=&pdtypeofsearch=TOCSearchDoc&pdfilterstring=MTA5MTIwMw&pdsearchdisplaytext=Official+Code+of+Georgia+Annotated&pdcontextvalue=statutes-legislation&pdtocfullpath=%2Fshared%2Ftableofcontents%2Furn%3AcontentItem%3A63RH-PW33-CH1B-T4TR-00008-00&pdbcts=1694467123349&config=00JABlN2Q2OTIwYi1kMjQwLTQxMWEtOWM1YS00MzUwY2MzYjQ5ZTAKAFBvZENhdGFsb2eiEwC2ZWq2J6k0Uwbdk8jZ&ecomp=7gf5kkk&prid=cfbd617a-9be3-4e6c-8a31-29248e6c555b
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:43]: add source (with sub paragraph) - https://law.justia.com/codes/georgia/2022/title-48/chapter-7/article-2/section-48-7-27/

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 20]
```diff
@@ -0,0 +1,46 @@
+description: List of Georgia adjusted gross income subtractions.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:1]:
```suggestion
description: Georgia subtracts the following from the adjusted gross income.
```

File: changelog_entry.yaml [hunk 21]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+      - Georgia adjusted gross income structure.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]: indent back

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml [hunk 22]
```diff
@@ -8,7 +8,7 @@ metadata:
     - title: 2022 Form 500 Instructions
       href: https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=12
     - title: 2021 Form 500 Instructions
-      href: https://houpl.org/wp-content/uploads/2022/01/2021_IT-511_Individual_Income_Tax_Booklet-12.29.21-Web-Version.pdf#page=13
+      href: https://www.zillionforms.com/2021/I2122607361.PDF#page21
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml:11]:
```suggestion
      href: https://www.zillionforms.com/2021/I2122607361.PDF#page=21
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+    # Income tax refunds
+    # Income from any fund
+    # Adjustment to Federal adjusted gross income
+    # Adjustment for teachers retired from the Teacher’s Re- tirement System of Georgia
+    # employersinfoodandbeverageestab- lishments
+    # minority subcontractors
+    # Path2College 529 Plan
+    # Adjustments due to Federal tax changes
+    # Combat zone pay exclusion
+    # axpayer’s donation of all or part of a kidney, liver, pancreas, intestine, lung or bone marrow during the taxable year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:22]: this comment might be too long

File: policyengine_us/variables/gov/states/ga/ga_agi.py [hunk 24]
```diff
@@ -7,6 +7,10 @@ class ga_agi(Variable):
     label = "Georgia adjusted gross income"
     unit = USD
     definition_period = YEAR
+    reference = (
+        "https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ga/ga_agi.py:11]:
```suggestion
        "https://houpl.org/wp-content/uploads/2023/01/2022-IT-511_Individual_Income_Tax_-Booklet-compressed.pdf#page=14",
```

File: policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+  2021-01-01:
+    - us_govt_interest
+    # ga_retirement_income_exclusion
+    # Military retirement income subtraction
+    # Dependent personal exemption for unborn children with a detectable human heartbeat
+    # Social security or railroad retirement
+    # Salaries and wages
+    # Individual retirement account, Keogh, SEP and SUB-S plan withdrawals
+    # Depreciation because of differences
+    # Dependent’s unearned income
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:12]:
```suggestion
    # Dependent's unearned income
```

Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:12]: i think the unicode windows error results from these apostrophes. try this here and elsewhere
```suggestion
    # Dependent's unearned income
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ga/tax/income/agi/subtractions.yaml:12]: this character does not work here, also in other parts of this file

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]
just needs another git pull upstream master - i think you did it during a separate pr merge

**Pull request merged**